soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -166,6 +166,36 @@ bool intel_adsp_ipc_send_message_sync(const struct device *dev,
 	return ret;
 }
 
+void intel_adsp_ipc_send_message_emergency(const struct device *dev, uint32_t data,
+					   uint32_t ext_data)
+{
+	const struct intel_adsp_ipc_config * const config = dev->config;
+
+	volatile struct intel_adsp_ipc * const regs = config->regs;
+	bool done;
+
+	/* check if host is processing message. */
+	while (regs->idr & INTEL_ADSP_IPC_BUSY)
+		k_busy_wait(1);
```

Comment:
Please add braces as required by our coding guidelines.

```suggestion
	while (regs->idr & INTEL_ADSP_IPC_BUSY) {
		k_busy_wait(1);
	}
```